src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1364,8 +1364,7 @@ uct_dc_mlx5_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
     UCT_CHECK_PARAM(comp == NULL, "Unsupported completion on ep_check");
     UCT_CHECK_PARAM(flags == 0, "Unsupported flags: %x", flags);
 
-    if (((ep->dci != UCT_DC_MLX5_EP_NO_DCI) &&
-         uct_dc_mlx5_iface_dci_has_outstanding(iface, ep->dci)) ||
```

Comment:
why removed?

Reply:
discussed with @brminich - check for has_outstanding is not needed

Reply:
if we have dci, it means we have something outstanding, right?

Reply:
@hoopoepg i thought that dci_put can avoid setting ep->dci=NO_DCI in some cases

Reply:
me too, but after some investigation we (with @brminich) found that this is not true